src/api/uploads/FolderUpload.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @flow
+ * @file Folder upload bootstrapping
+ * @author Box
+ */
+
+import FolderUploadNode from './FolderUploadNode';
+
+const PATH_DELIMITER = '/';
+
+class FolderUpload {
+    folderNodes: Object = {};
+    files: Array<File> = [];
+    destinationFolderID: string;
+    createFolder: Function;
+    uploadFile: Function;
+    addFolderToQueue: Function;
+
+    /**
+     * [constructor]
+     *
+     * @param {Function} createFolder
+     * @param {Function} uploadFile
+     * @param {string} destinationFolderID
+     * @param {Function} addFolderToQueue
+     * @return {void}
+     */
+    constructor(
+        createFolder: Function,
+        uploadFile: Function,
+        destinationFolderID: string,
+        addFolderToQueue: Function
+    ): void {
+        this.createFolder = createFolder;
+        this.uploadFile = uploadFile;
+        this.destinationFolderID = destinationFolderID;
+        this.addFolderToQueue = addFolderToQueue;
+    }
+
+    /**
+     * Creates a folder tree from wekbkitRelativePath
+     *
+     * @param  {FileList} Array<UploadFileWithAPIOptions | File> | FileList
+     * @returns {void}
+     */
+    buildFolderTree(fileList: Array<UploadFileWithAPIOptions | File> | FileList): void {
```

Comment:
probably port over the test for this too

Reply:
yeah writing tests